### PR TITLE
Hadamard bound: double instead of size_t for bitSize

### PR DESCRIPTION
**Pull request by Breush**

### CODE REVIEW

File: linbox/algorithms/lifting-container.h [hunk 1]
```diff
@@ -152,12 +152,15 @@ namespace LinBox
...
-			L = N * D * 2;
-
-			_length = (size_t)logp(L,Prime) + 1;   // round up instead of down
+            auto hb = RationalSolveHadamardBound(A, b);
+            N = Givaro::pow(Integer(2), static_cast<size_t>(std::ceil(hb.numLogBound)));
+            D = Givaro::pow(Integer(2), static_cast<size_t>(std::ceil(hb.denLogBound)));
+
+            // L = N * D * 2
+            // _length = logp(L, Prime) = log2(L) * ln(2) / ln(Prime)
+            double primeLog = Givaro::naturallog(Prime);
```
Comment (ClementPernet) [inline linbox/algorithms/lifting-container.h]: why take the natural log if you really want the log2 (line below divides by log(2)). There is a log_2 function in givaro (gmp++)

File: linbox/algorithms/lifting-container.h [hunk 2]
```diff
@@ -152,12 +152,15 @@ namespace LinBox
...
             this->_intRing.convert(Prime,_p);
 
-            auto hadamardBound = RationalSolveHadamardBound(A, b);
-            N = Givaro::pow(Integer(2), hadamardBound.numBoundBitSize);
-            D = Givaro::pow(Integer(2), hadamardBound.denBoundBitSize);
-			L = N * D * 2;
-
-			_length = (size_t)logp(L,Prime) + 1;   // round up instead of down
+            auto hb = RationalSolveHadamardBound(A, b);
+            N = Givaro::pow(Integer(2), static_cast<size_t>(std::ceil(hb.numLogBound)));
```
Comment (ClementPernet) [inline linbox/algorithms/lifting-container.h]: Maybe you can directly replace the pow by the 1<< here (and not in a future PR)

File: linbox/algorithms/lifting-container.h (hunk 1 above)
Comment (ClementPernet) [reply linbox/algorithms/lifting-container.h]: or Givaro::logtwo which you are using later on...

File: linbox/algorithms/lifting-container.h (hunk 2 above)
Comment (Breush) [reply linbox/algorithms/lifting-container.h]: Forgot that, thanks. Fixed

File: linbox/algorithms/lifting-container.h (hunk 1 above)
Comment (Breush) [reply linbox/algorithms/lifting-container.h]: Updated.

**Comment by ClementPernet** [comment]
LGTM

**Pull request merged**